### ISSUE TITLE
Cache

### ISSUE THREAD

**Issue by daslicht** [gmrchk/swup#45]
Hi, 
The cache is at the moment just in memory, correct?
How about persisting it into the localStorage?

Regarding chnages in content,
Probably it is enough to add a class which indicates the current state ,
Eg version–1 etc
On load just compare the version with the cache :)

**Comment by gmrchk** [comment]
Hi, 

Good idea! I had similar some time ago, only with meta tag instead of class.

```html
<meta name="swup-cache" content="[any string]" />
```

Unfortunately, it's a bit tricky. You would ideally have to discard the saved cache in localStorage everytime you change anything (css, js, the structure of your website, etc.). This could be automatized with (let's say) writing the time of last change of all files you depend on into that meta tag, so anytime one of the files changes, the cache gets discarded.  

There is another thing with dynamic sites (blogs, etc.), where the time of the last update would have to be taken into account as well.

Long story short, I don't think it's something that should be included in swup, as it's kind of hard to implement in most cases and probably wouldn't bring that much of an improvement. However, since last time swup gained a support for plugins, and I think this would be a perfect material for one.

What do you think?

**Comment by gmrchk** [comment]
I guess this feature is not really desired by the community. I will close this for now, but if anyone would like to see this happen, let me know.

**Comment by frederikhors** [comment]
It's still an amazing, killer feature. Please re-open, @gmrchk.

**Comment by gmrchk** [comment]
I would actually suggest going for a [service worker](https://developers.google.com/web/fundamentals/primers/service-workers/) solution. It's native, it can load even the first page when offline, it will improve your site rating, and there are different approaches that you can take like loading the cached or live version first.

**Comment by daslicht** [comment]
hm isn't that overkill compared to just store things in a localStorage.?
